main/php_streams.h

```diff
@@ -287,11 +287,21 @@ END_EXTERN_C()
 
 BEGIN_EXTERN_C()
 
-static zend_always_inline bool php_stream_zend_parse_arg_into_stream(zval *arg, php_stream **destination_stream_ptr, bool check_null)
-{
+static zend_always_inline bool php_stream_zend_parse_arg_into_stream(
+	zval *arg,
+	php_stream **destination_stream_ptr,
+	bool check_null,
+	uint32_t arg_num
+) {
 	if (EXPECTED(Z_TYPE_P(arg) == IS_RESOURCE)) {
-		*destination_stream_ptr = (php_stream*)zend_fetch_resource2(Z_RES_P(arg), "stream", php_file_le_stream(), php_file_le_pstream());
-		if (UNEXPECTED(*destination_stream_ptr == NULL)) {
+		zend_resource *res = Z_RES_P(arg);
+		/* We do not use zend_fetch_resource2() API,
+		 * as we want to be able to specify the argument number in the type error */
+		if (EXPECTED(res->type == php_file_le_stream() || res->type == php_file_le_pstream())) {
+			*destination_stream_ptr = (php_stream*)res->ptr;
+			return true;
+		} else {
+			zend_argument_type_error(arg_num, "must be an open stream resource");
```

Comment:
open vs valid? The existing code uses valid instead of open, and this should be consistent imo.
Other parts of the PR look fine I think.

Reply:
The problem I find is that a closed stream still feels "valid" but will fail such a check. This is why I changed it to "open"